### PR TITLE
DACT-1714 Obfuscate residential address if officer is secure

### PR DESCRIPTION
**Pull request by jditchburn**
https://companieshouse.atlassian.net/browse/DACT-1714

### CODE REVIEW

**Comment by jditchburn** [comment]
Sonar says I am missing a test scenario but I'm not sure I am. Maybe someone can see the scenario that's missing?

File: test/services/company.appointments.service.unit.ts [hunk 1]
```diff
@@ -91,5 +91,42 @@ describe("Company Appointments service test", () => {
...
+      expect(returnedAppointment.usualResidentialAddress.careOf).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.country).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.locality).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.poBox).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.postalCode).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.premises).toEqual("********");
+      expect(returnedAppointment.usualResidentialAddress.region).toEqual("********");
+    });
+
+    test.each([false, null])
```
Comment (mdesaikns) [inline test/services/company.appointments.service.unit.ts:113]: regarding your comment
`Sonar says I am missing a test scenario but I'm not sure I am. Maybe someone can see the scenario that's missing?`

try adding `undefined` in `test.each[]` array, if I am not wrong the null and undefined are slightly different.

Comment (jditchburn) [reply test/services/company.appointments.service.unit.ts]: Didn't help unfortunately

**Review by mdesaikns — APPROVED** [review]

**Comment by AndreKramer7654** [comment]
Drop the "?" on 
if (sdkResponse.resource?.isSecureOfficer) {
It's bound to be defined as there's a test a couple of lines up. 
Hopefully that will get you to 100% coverage.

File: src/services/company.appointments.service.ts [hunk 2]
```diff
@@ -31,5 +31,23 @@ export const getCompanyAppointmentFullRecord = async (session: Session, companyN
 
   logger.debug(`Received company appointment ${JSON.stringify(sdkResponse)}`);
 
+  if (sdkResponse.resource?.isSecureOfficer) {
+    obfuscateResidentialAddress(sdkResponse.resource);
+  }
+
   return sdkResponse.resource;
 };
+
```
Comment (AndreKramer7654) [inline src/services/company.appointments.service.ts:40]: consider using 
const obfuscatedPattern = "*******";

**Review by AndreKramer7654 — APPROVED** [review]

**Review by clewis1 — APPROVED** [review]

**Pull request merged**